### PR TITLE
docs(headless): add React sample for standalone search box

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-422

### CODE REVIEW

File: packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  private getURIParameters() {
+    const {redirectTo, value} = this.state;
+    const query = encodeURIComponent(value);
+    // The query pipeline can trigger different redirection URLs, as documented here: https://docs.coveo.com/en/1458/tune-relevance/trigger-query-pipeline-feature
+    // Unless a redirect trigger is added, you won't need such conditions. These redirections are only verified here as an example.
```
Comment (btaillon) [inline packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx:48]: I'm not sure what to write here... Opinions?

@fbeaudoincoveo @kaitlynkatz @jpmarceau

File: packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.fn.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    const {redirectTo, value} = state;
+    const query = encodeURIComponent(value);
+    // The query pipeline can trigger different redirection URLs, as documented here: https://docs.coveo.com/en/1458/tune-relevance/trigger-query-pipeline-feature
+    // Unless a redirect trigger is added, you won't need such conditions. These redirections are only verified here as an example.
+    if (redirectTo === 'https://mywebsite.com/search') {
+      return '/' + query;
+    } else if (redirectTo === 'https://www.google.com') {
+      return '?q=' + query;
+    }
+    return '?search=' + query;
```
Comment (samisayegh) [inline packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.fn.tsx:26]: Why are we checking the `redirectTo` url in the sample? Why not just use the `redirectTo` in state?

I really believe the goal with these samples should be to show a simple use-case with the controller.

When a sample is simple, people will think of headless as simple. When a sample is complex, headless will be perceived as complicated, which would not be a fair representation.

**Review by samisayegh — APPROVED** [review]

File: packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.fn.tsx (hunk 2 above)
Comment (btaillon) [reply packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.fn.tsx:26]: I'm checking it for the reason I wrote above.

However, it's not strictly necessary and I agree, it does make the sample look more complicated than necessary. I can remove it

File: packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx (hunk 1 above)
Comment (olamothe) [reply packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx:48]: I would vote to remove that explanation for the sample that are meant to appear in the controller documentation.

Those more complex "how to actually do redirection properly from A to Z with headless + query pipeline + trigger" and what not, I would keep for an "advanced usage" section of the documentation.

And, as Sami commented, to remove the `redirectTo` check for both example.

**Review by olamothe — APPROVED** [review]

File: packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx (hunk 1 above)
Comment (jpmarceau) [reply packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx:48]: I agree 🙂👍

File: packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  componentDidUpdate() {
+    if (!this.state?.redirectTo) {
+      return;
+    }
+    window.location.href = `${this.state.redirectTo}/${encodeURIComponent(
+      this.state.value
```
Comment (ThibodeauJF) [inline packages/samples/headless-react/src/components/standalone-search-box/standalone-search-box.class.tsx:38]: I would remove the encoded part and just redirect to the exact redirection url in order not to mislead the user

**Pull request merged**